Tools/px4moduledoc/markdownout.py

```diff
@@ -40,25 +40,27 @@ def __init__(self, module_groups):
 ## Categories
 """
         for category in sorted(module_groups):
-            result += "- [%s](modules_%s.md)\n" % (category.capitalize(), category)
+            result += f"- [{category.capitalize()}](modules_{category}.md)\n"
 
         self._outputs['main'] = result
 
         for category in sorted(module_groups):
-            result = "# Modules Reference: %s\n" % category.capitalize()
+            result = f"# Modules Reference: {category.capitalize()}\n\n"
             subcategories = module_groups[category]
+
             if len(subcategories) > 1:
-                result += 'Subcategories:\n'
-                for subcategory in subcategories:
+                sorted_subcategories = {key: subcategories[key] for key in sorted(subcategories.keys())}
+                result += 'Subcategories:\n\n'
+
+                for subcategory in sorted_subcategories:
```

Comment:
This can be simplified a bit:
```suggestion
                for subcategory in sorted(subcategories):
```

Reply:
@bkueng I thought so too, but `sorted()` implicitly converts the object into an array. It's not the same object which makes  the program crash further on down.

Reply:
does this work?

```suggestion
                for subcategory in dict(sorted(subcategories.items())):
```

Reply:
Ok weird, `subcategory` should still be a string though.

Using `{key: subcategories[key] for key in sorted(subcategories.keys())}` isn't guaranteed to be ordered when iterating, as it's still a dictionary.

Reply:
@bkueng @mrpollo 
1. A dict is ordered in addition order from Python 3.6 (or maybe 7). We're using greater than that everywhere.
2. On retesting, all the methods work, including Beat's suggestion. I don't know why it didn't work when I tried it before.

Accepting @bkueng suggestion as "the simplest"